### DISCUSSION TITLE
OIDC Client Secret parameter (Google Oauth2)

### DISCUSSION THREAD

**Discussion by GravityHurts** [mealie-recipes/mealie#3805] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Today, we can set up OIDC providers, but cannot pass a "client_secret" parameter.

### Please provide a concise description of the feature that would resolve your issue.

This would allow users to use Google as an OIDC/Oauth2 provider. Today, trying to set up Google's implementation of Oauth2 results in an error (missing client_secret)

### Please consider and list out some caveats or tradeoffs made in your design decision

None that I'm aware of- unsure of how to implement this

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [x] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by GravityHurts** [comment]
Follow-up:

I managed to get google account based auth set up with Auth0, so I'm able to sign in with a google account/sso... Still curious if this is an easy change to add the "client_secret" parameter on the oidc token authorization call to the provider.

**Reply by cmintey** [reply]
Glad you got it working with Auth0. As for why google doesn't work directly, this is a limitation with them and Mealie's particular auth scheme. See my reply to a similar question here: https://github.com/mealie-recipes/mealie/discussions/3477

**Comment by GravityHurts** [comment]
I understand that we cannot support this.

I'd recommend others that wish to sign in with google try out Auth0 - it was relatively easy to set up.
